src/user-data.ts

```diff
@@ -131,7 +131,8 @@ export type AppCourse = {
   readonly color: string;
   readonly check: boolean;
   uniqueID: number;
-  readonly isReqCourse: boolean;
+  isReqCourse: boolean;
+  isCompletedReqCourse: boolean;
```

Comment:
why these two fields have to be mutable?
I'm also somehow against the idea of adding more field to this course object. This object is already very bloated and it's becoming a do-it-all object. Can we avoid this?

Reply:
@SamChou19815 For now, I think that we may have to change the value of the fields when the course is dragged from the requirements bar to the schedule (e.g. so that the triple dots appear on the course card). I will have a better answer when I work on my PR for this week but I will change to readonly if my assumption is incorrect.

As for adding `isCompletedReqCourse`, my reasoning was to add a param to style the Course component for this special case. An option that we can explore is to just create a div in `CompletedSubReqCourse.vue` that looks like a compact, isCompletedReqCourse Course card without actually using the Course component.

Reply:
It looks like this field is only useful for certain components, so why can't we pass `isReqCourse` and `isCompletedReqCourse` along with an `AppCourse` in those components, instead of merging all of them together.

Reply:
@SamChou19815 For now, I will make a separate component for a (completed) requirement course (see screenshot below), which will remove the need for an `isCompletedReqCourse` field. I will do this in a following PR.
![Screen Shot 2020-12-01 at 8 25 38 PM](https://user-images.githubusercontent.com/54298311/100816091-60eb7a00-3413-11eb-8dc8-b90c737a4bb1.png)

I will have to do more exploring, but I might be able to remove the need for an `isReqCourse` field if the new requirement course component also handles a nontaken requirement course (see below). As mentioned in our dev meeting, this might involve a lot of refactoring so this may not be the immediate solution.
![Screen Shot 2020-12-01 at 8 27 29 PM](https://user-images.githubusercontent.com/54298311/100816227-a3ad5200-3413-11eb-867c-22cc696db2a3.png)